src/pm-conversations/pmConversationsSelectors.js

```diff
@@ -65,9 +72,50 @@ export const getRecentConversationsLegacy: Selector<PmConversationData[]> = crea
   },
 );
 
-export const getRecentConversationsModern: Selector<PmConversationData[]> = state =>
-  // TODO implement for real
-  getRecentConversationsLegacy(state);
+export const getRecentConversationsModern: Selector<PmConversationData[]> = createSelector(
+  state => state.pmConversations,
+  getUnreadByPms,
+  getUnreadByHuddles,
+  getAllUsersById,
+  getOwnUserId,
+  // This is defined separately, just below.  When this is defined inline,
+  // if there's a type error in it, the message Flow gives is often pretty
+  // terrible: just highlighting the whole thing and pointing at something
+  // in the `reselect` libdef.  Defining it separately seems to help.
+  getRecentConversationsModernImpl, // eslint-disable-line no-use-before-define
+);
+
+function getRecentConversationsModernImpl(
+  { sorted, map }: PmConversationsState,
+  unreadPms,
+  unreadHuddles,
+  allUsersById,
+  ownUserId,
+): PmConversationData[] {
+  return sorted
+    .toSeq()
+    .map(recentsKey => {
+      const keyRecipients = pmKeyRecipientsFromIds(
+        model.usersOfKey(recentsKey),
+        allUsersById,
+        ownUserId,
+      );
+      if (keyRecipients === null) {
```

Comment:
That's what `pmKeyRecipientsFromIds` returns if there's a user ID it can't find in the given `allUsersById` map. By returning null, it forces (via the type-checker) its caller to deal with it -- either by throwing, or as I've done here by ignoring the conversation and moving on.

I see its jsdoc doesn't mention that, though! I'll add that.

Reply:
> That's what `pmKeyRecipientsFromIds` returns if there's a user ID it can't find in the given `allUsersById` map.

True, but I guess I was thinking, why would there be a user ID that can't be found in the given `allUsersById` map? 😛

Reply:
Probably shouldn't be possible :slightly_smiling_face: -- there aren't even the scenarios there are for emails, where someone may have changed their email. So probably we should emit a warning to Sentry from `pmKeyRecipientsFromIds` in this case.

If it *does* happen, then at least at this callsite of `pmKeyRecipientsFromIds` it seems pretty clear how best to proceed: ignore that one convesration and carry on. If there weren't a clear reasonable way to proceed, then we'd want to throw instead.